### DISCUSSION TITLE
Foonathan Memory error when compiling Fast-DDS：The following configuration files were considered but not accepted

### DISCUSSION THREAD

**Discussion by Vermounthlly** [eProsima/Fast-DDS#2781] [Q&A]
I have installed Fast-CDR ，ASIO， foonathan_memory_vendor，but can't install Fast-DDS

This is the Terminal error：
`- Setting build type to 'Release' as none was specified.
-- The C compiler identification is GNU 4.9.4
-- The CXX compiler identification is GNU 4.9.4
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/arm/gcc-linaro-4.9.4-2017.01-x86_64_arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/arm/gcc-linaro-4.9.4-2017.01-x86_64_arm-linux-gnueabihf/bin/arm-linux-gnueabihf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Version: 2.7.0
-- Performing Test SUPPORTS_CXX20
-- Performing Test SUPPORTS_CXX20 - Failed
-- Performing Test SUPPORTS_CXX17
-- Performing Test SUPPORTS_CXX17 - Failed
-- Performing Test SUPPORTS_CXX14
-- Performing Test SUPPORTS_CXX14 - Success
-- Performing Test SUPPORTS_CXX1Y
-- Performing Test SUPPORTS_CXX1Y - Success
-- Performing Test SUPPORTS_CXX11
-- Performing Test SUPPORTS_CXX11 - Success
-- Found fastcdr
-- Found Asio 1.22.1: /usr/local/arm/gcc-linaro-4.9.4-2017.01-x86_64_arm-linux-gnueabihf/include
-- Found Asio
-- Found TinyXML2: /home/lly/Projects/Fast-DDS/arm-install/lib/cmake/tinyxml2
-- Found TinyXML2
CMake Error at CMakeLists.txt:182 (find_package):
  Could not find a configuration file for package "foonathan_memory" that is
  compatible with requested version "".

  The following configuration files were considered but not accepted:

    /home/lly/Projects/Fast-DDS/arm-install/lib/foonathan_memory/cmake/foonathan_memory-config.cmake, version: 0.7.1 (64bit)

-- Configuring incomplete, errors occurred!
See also "/home/lly/Projects/Fast-DDS/Fast-DDS/arm_build/CMakeFiles/CMakeOutput.log".`

Can anyone suggest what mistake I'm doing in these steps. Thanks！！！！

**Comment by JLBuenoLopez** [comment]
Hi @Vermounthlly,

Could you please provide us more information about your environment? Reading the traces, it seems that you are building for an ARM architecture... Are you crosscompiling?

**Reply by Vermounthlly** [reply]
Thank you for your reply!!
Yes,I'm trying to cross compile Fast-DDS using x86_64_arm-linux-gnueabihf-gcc-4.9.0 ,and got that error.

**This is System information:
    Ubuntu 18.04
    Fast-DDS Version 2.7.0
    Asio Version 1.22.1**

By the way, I got this error, and then I changed the cross-compiler------ aarch64-linux-gnu-gcc version 8.4.0
also has another error:
<pre><font color="#8AE234"><b>lly@ubuntu</b></font>:<font color="#729FCF"><b>~/Fast-DDS/Fast-DDS/aarch64_build</b></font>$ cmake --build . --target install
[  1%] <font color="#4E9A06">Building CXX object src/cpp/CMakeFiles/fastrtps.dir/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.cpp.o</font>
In file included from <b>/usr/include/wchar.h:30</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/cwchar:44</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/bits/postypes.h:40</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/iosfwd:40</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/memory:72</b>,
                 from <b>/home/lly/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.hpp:22</b>,
                 from <b>/home/lly/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.cpp:19</b>:
<b>/usr/include/bits/floatn.h:87:9:</b> <font color="#EF2929"><b>error: </b></font>‘<b>__float128</b>’ does not name a type; did you mean ‘<b>__cfloat128</b>’?
 typedef <font color="#EF2929"><b>__float128</b></font> _Float128;
         <font color="#EF2929"><b>^~~~~~~~~~</b></font>
         <font color="#4E9A06">__cfloat128</font>
In file included from <b>/usr/aarch64-linux-gnu/include/c++/8/cwchar:44</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/bits/postypes.h:40</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/iosfwd:40</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/memory:72</b>,
                 from <b>/home/lly/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.hpp:22</b>,
                 from <b>/home/lly/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.cpp:19</b>:
<b>/usr/include/wchar.h:406:8:</b> <font color="#EF2929"><b>error: </b></font>‘<b>_Float128</b>’ does not name a type; did you mean ‘<b>_Float32x</b>’?
 extern <font color="#EF2929"><b>_Float128</b></font> wcstof128 (const wchar_t *__restrict __nptr,
        <font color="#EF2929"><b>^~~~~~~~~</b></font>
        <font color="#4E9A06">_Float32x</font>
<b>/usr/include/wchar.h:523:8:</b> <font color="#EF2929"><b>error: </b></font>‘<b>_Float128</b>’ does not name a type; did you mean ‘<b>_Float32x</b>’?
 extern <font color="#EF2929"><b>_Float128</b></font> wcstof128_l (const wchar_t *__restrict __nptr,
        <font color="#EF2929"><b>^~~~~~~~~</b></font>
        <font color="#4E9A06">_Float32x</font>
In file included from <b>/usr/aarch64-linux-gnu/include/c++/8/cstdlib:75</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/ext/string_conversions.h:41</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/bits/basic_string.h:6400</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/string:52</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/stdexcept:39</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/array:39</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/tuple:39</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/bits/unique_ptr.h:37</b>,
                 from <b>/usr/aarch64-linux-gnu/include/c++/8/memory:80</b>,
                 from <b>/home/lly/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.hpp:22</b>,
                 from <b>/home/lly/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.cpp:19</b>:
<b>/usr/aarch64-linux-gnu/include/stdlib.h:152:8:</b> <font color="#EF2929"><b>error: </b></font>‘<b>_Float128</b>’ does not name a type; did you mean ‘<b>_Float32x</b>’?
 extern <font color="#EF2929"><b>_Float128</b></font> strtof128 (const char *__restrict __nptr,
        <font color="#EF2929"><b>^~~~~~~~~</b></font>
        <font color="#4E9A06">_Float32x</font>
<b>/usr/aarch64-linux-gnu/include/stdlib.h:245:4:</b> <font color="#EF2929"><b>error: </b></font>‘<b>_Float128</b>’ has not been declared
    <font color="#EF2929"><b>_Float128</b></font> __f)
    <font color="#EF2929"><b>^~~~~~~~~</b></font>
<b>/usr/aarch64-linux-gnu/include/stdlib.h:330:8:</b> <font color="#EF2929"><b>error: </b></font>‘<b>_Float128</b>’ does not name a type; did you mean ‘<b>_Float32x</b>’?
 extern <font color="#EF2929"><b>_Float128</b></font> strtof128_l (const char *__restrict __nptr,
        <font color="#EF2929"><b>^~~~~~~~~</b></font>
        <font color="#4E9A06">_Float32x</font>
In file included from <b>/home/lly/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.hpp:22</b>,
                 from <b>/home/lly/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.cpp:19</b>:
<b>/usr/aarch64-linux-gnu/include/c++/8/memory:</b> In function ‘<b>void* std::align(std::size_t, std::size_t, void*&amp;, std::size_t&amp;)</b>’:
<b>/usr/aarch64-linux-gnu/include/c++/8/memory:116:58:</b> <font color="#EF2929"><b>error: </b></font>cast from ‘<b>void*</b>’ to ‘<b>uintptr_t</b>’ {aka ‘<b>unsigned int</b>’} loses precision [<font color="#EF2929"><b>-fpermissive</b></font>]
   const auto __intptr = reinterpret_cast&lt;uintptr_t&gt;(__ptr<font color="#EF2929"><b>)</b></font>;
                                                          <font color="#EF2929"><b>^</b></font>
src/cpp/CMakeFiles/fastrtps.dir/build.make:75: recipe for target &apos;src/cpp/CMakeFiles/fastrtps.dir/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.cpp.o&apos; failed
make[2]: *** [src/cpp/CMakeFiles/fastrtps.dir/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.cpp.o] Error 1
CMakeFiles/Makefile2:940: recipe for target &apos;src/cpp/CMakeFiles/fastrtps.dir/all&apos; failed
make[1]: *** [src/cpp/CMakeFiles/fastrtps.dir/all] Error 2
Makefile:145: recipe for target &apos;all&apos; failed
make: *** [all] Error 2
</pre>

and this is a part of CMakeLists.txt:

`cmake_minimum_required(VERSION 3.5)
add_definitions(-std=c++11)
cmake_policy(VERSION 3.5...3.13)
if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
    message(STATUS "Setting build type to 'Release' as none was specified.")
    set(CMAKE_BUILD_TYPE Release CACHE STRING
        "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel."
        FORCE)
    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
endif()
set(CMAKE_STSTEM_NAME Linux)
set(CMAKE_SYSROOT "/usr/bin/")
set(tool_path "/usr/bin/")
set(CMAKE_CROSSCOMPILING TRUE)
set(CMAKE_FIND_ROOT_PATH ${tool_path})
set(CMALE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
set(CMALE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
set(CMALE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
set(CMALE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
set(CMAKE_C_COMPILER ${tool_path}aarch64-linux-gnu-gcc-8)
set(CMAKE_CXX_COMPILER ${tool_path}aarch64-linux-gnu-g++-8)
set(CMAKE_LINKER ${tool_path}aarch64-linux-gnu-ld)
project(fastrtps VERSION "2.7.0" LANGUAGES C CXX)`


I don't know how to solve these  problems......

**Reply by Vermounthlly** [reply]
I have fixed the problem. I used the wrong compiler version
